elements/progress-slider/progress-slider.js

```diff
@@ -123,11 +125,24 @@ let self = class ProgressSlider extends HTMLElement {
 		}
 
 		if (name === "show") {
+			if (newValue === "progress") {
+				this.valueElement.setAttribute("min", 0);
+				this.valueElement.setAttribute("max", 1);
 
+				// `step` should be proportional to the one the component already has.
+				this.valueElement.setAttribute("step", this.step / (this.max - this.min));
+			}
+			else {
+				["min", "max", "step"].forEach(prop => this.valueElement.setAttribute(prop, this[prop]));
+			}
 		}
 		else {
 			this.sliderElement.setAttribute(name, newValue);
 			this.valueElement.setAttribute(name, newValue);
+
+			if (["min", "max", "value"].includes(name)) {
+				this.style.setProperty("--progress", this.progress);
```

Comment:
`--progress` should be set regardless of the value of `show`. But also, all the side effects after min, max, step changing should move somewhere else and just be called (perhaps a `#constraintsUpdated()` function?)

Reply:
>  `--progress` should be set regardless of the value of `show`.

Do you think it should be set even though none of its dependencies is updated (like in this case)?